### ISSUE TITLE
textDocument/hover: escape html entities for markdown

### ISSUE THREAD

**Issue by rwols** [microsoft/pyright#1082]
**Describe the bug**
The textDocument/hover response advertises markdown, but doesn't escape `<` and `>` characters.

**To Reproduce**
Create any docstring with e.g. the string `<code>` in it.

**Expected behavior**
pyright should escape html entities for consumption in markdown.

**Screenshots or Code**
```python
class Foo:
    """Works around <code> tags in html"""
    pass
```

Request and response payloads:
```
:: --> LSP-pyright textDocument/hover(1146): {'textDocument': {'uri': 'file:///Users/raoulwols/dev/LSP-SonarLint/plugin.py'}, 'position': {'line': 21, 'character': 8}}
:: <<< LSP-pyright 1146: {'range': {'end': {'line': 21, 'character': 9}, 'start': {'line': 21, 'character': 6}}, 'contents': {'value': '```python\n(class) Foo\n```\nWorks around <code> tags in html', 'kind': 'markdown'}}
```

**VS Code extension or command-line**
N/A

**Additional context**
To be honest, I am wondering whether pyright sending the docstring "as-is" is correct. Because docstyles in Python do not adhere to a definite standard, perhaps wrap the entire docstring in triple backticks to maintain indentation. Because indentation is really the only thing docstrings have going for them with respect to style.

**Comment by jakebailey** [comment]
> To be honest, I am wondering whether pyright sending the docstring "as-is" is correct. Because docstyles in Python do not adhere to a definite standard, perhaps wrap the entire docstring in triple backticks to maintain indentation. Because indentation is really the only thing docstrings have going for them with respect to style.

Pyright/pylance don't send it as-is, there's a converter that attempts to handle various types of docstrings (reStructuredText, markdown, etc). It could be modified to ensure that HTML tags were escaped in the non-code blocks (which it already attempts to distinguish).

Converting docstrings to markdown consistently is hard as there's no single format. We can't just put docstrings into code blocks, as this renders poorly: the docstring has line wrapping (inserted by the author) and the editor will also wrap as needed to fit the text, which combined leads to awkward double line wraps.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by rwols** [comment]
Thanks, you guys are *on-point* with releases!
